### ISSUE TITLE
ISO image made with Customizer boots live session and installs but doesn't boot installed system

### ISSUE THREAD

**Issue by UriHerrera** [kamilion/customizer#88]
Running Customizer 3.2.3 

Selected ubuntu-mini-remix-14.04.1-amd64.iso - The customized ISO file builds and boots correctly to the live session in addition it seemingly installs the system too without a problem, however the resulting installation doesn't boot.

I'm not sure if it's a missing packages problem or it's the UMR image I'm using. Any help would be appreciated.

**Comment by fluxer** [comment]
v3.2.3 is not supposed to be used with Ubuntu 12.04+. Sharing the ISO image would be nice so I do not have to guess what's the problem and can try it out myself, if possible.

**Comment by UriHerrera** [comment]
Sure, just let me upload the ISO somewhere. What version of Customizer should I use instead?.

**Comment by fluxer** [comment]
You can grab 4.1.0 from https://github.com/clearkimura/Customizer/releases or the latest version from https://github.com/clearkimura/Customizer/archive/master.tar.gz.

**Comment by UriHerrera** [comment]
Here's the link to the file: https://www.dropbox.com/s/4ntyz952h0b7w31/Nitrux-amd64-2.15-kernel-4.0rc3.iso?dl=0

**Comment by fluxer** [comment]
It seems that it works fine, I tried installing it in virtualbox. Maybe someone else wants to try it out and see if he can reproduce your issue.

**Comment by UriHerrera** [comment]
I just tried to install the ISO on actual hardware but it also happens. As far as having a LiveCD the ISO is perfect, but it can't be installed, or rather the installation doesn't complete.

Ubiquity crashes but no error message is displayed. It stops after "Checking the CD-ROM" and right before the process of installing the bootloader.

I have tried something else too, I've tried to use that ISO with version 4.1.0 and it wouldn't work. Also I've noticed that the installer says it requires at least 1.6 GB of available space, strange. The installation should be around 6 GB. On that note, as I mentioned 4.1.0 wouldn't unsquash the ISO so I used 3.2.3 and then built it with 4.1.0, in the resulting ISO Ubiquity doesn't give the user the option to "Install or Try" it simply starts the Live session, not a problem exactly. The installer also reports that the required space is now 5.4 GB something that seems more accurate to me, but strangely enough it says the computer is not connected to the internet, which would be weird as Chromium does load pages but the virtual terminals and Konsole can't ping anything and can't connect to the repos.

This ISO (built with 4.1.0) does install the system, the bootloader and the resulting installation on actual hardware boots and so does in Virtualbox (unlike the one built with 3.2.3), however, no matter what, X crashes and of course, there's no network connection for anything related to a terminal.

So the tl;d version would be:
- ISO built with Customizer 3.2.3 works as a LiveCD but it can't be installed
- ISO built with Customizer 4.1.0 works as a LiveCD and can be installed but X crashes every time and there's no network connection (apparently).

**Comment by fluxer** [comment]
The problem with the disk space requirement on installation is (somewhat) fixed with d713647.

The problem with UMR ISO images is that they do not follow the structure of newer release (or at least 12.04-13.x IIRC) - casper/filesystem.manifest-remove has been renamed to casper/filesystem.manifers-desktop in the official ISOs since 12.04+. I stated that UMR is not a good solution (IMO) anymore in #76 for minimal remixes. UMR is something I personally do not want to support but if you do then I'm fine with it, patches welcome.

The only difference in the exclude file between 3.2.3 and 4.x is 11c5e49, nothing drastic. I suspected that it is to blame as it tells mksquashfs to exclude files such as etc/resolv.conf which may be to cuase. I suggest that you try to play with the file (for 4.x it should be installed as /usr/share/customizer/exclude.list). I have had mixed reports from users that the file should be/not be excluded. Different network managers (wicd, network-manager, etc.) handle things differently and making them happy is not easy. The problem with etc/resolve.conf in particular is that it is copied from the host to have network access inside the chroot but if kept it (usually) causes problems on with the LiveCD so it is excluded by default to prevent the bigger problem.

X craching - that's something new to me when it comes to remixes. Is it only the customized issue that you encounter this with or is it reproducable with official releases too? Note that matching your remix release and architecture with the official ones to test this would be cruicial as a lot of software (X, Mesa, Kernel, etc.) get updated between releases that have their own set of problems.

Sadly, I can not test it on bare metal. I hate to mess with my partitions and I have none to spare on both of my machines. I will try to assist if I can but apparently I'm not able to help much. If you have more findings on the issue do share them.

Cheers!
